api/src/wfl/module/covid.clj

```diff
@@ -303,6 +231,134 @@
         (assoc :type terra-workspace-sink-type))
     (throw (ex-info "Invalid sink_items" details))))
 
+(defn ^:private add-workload-record
```

Comment:
Ah. :) 

I know that you moved this code around to verify against `tdr-source-name` rather than a hardcoded name at my suggestion, but consider the following about the block of moved code from L234 - L361:

- It's all been moved to live in the implied "Terra Data Repository Source" section
- It contains general utilities, general COVID workload functionality, and validators for the `executor` and `sink` as well as the `source`
- The validators for the `executor` and the `sink` still check hard-coded names

This PR has been through a lot!  It might make most sense to move this code back to where you found it and stick with the hard-coding.  I'd like to have a team discussion about our desired structure for this namespace as it coalesces, and that may illuminate the path forward here.

Or, @ehigham might have another actionable suggestion.

Reply:
I tried to move the workflow functions into their own section below the TDR repository, so that the create, start, etc... which I labelled "Workload functions". I believe that the sink was already using the variable from the TDR sink section, but I also just updated the Executor to also use the TDR defn. I can revert if we want that. Please let me know.

Reply:
My apologies, I misinterpreted the git diff view!  Viewing the whole file helped me understand better.

Just two smaller requests then, which I'll put in comments inline below.

Reply:
No worries - I really appreciate the attention that you and everyone else has given to this! It's a small part of the project, but it has helped me learn so much. Thank you!